logfire/_internal/client.py

```diff
@@ -0,0 +1,102 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+from urllib.parse import urljoin
+
+from requests import Response, Session
+
+from logfire.exceptions import LogfireConfigError
+from logfire.version import VERSION
+
+from .utils import UnexpectedResponse
+
+if TYPE_CHECKING:
+    from .auth import UserToken
+
+
+UA_HEADER = f'logfire/{VERSION}'
+
+
+class ProjectAlreadyExists(Exception):
+    pass
+
+
+class InvalidProjectName(Exception):
+    def __init__(self, reason: str, /) -> None:
+        self.reason = reason
+
+
+class LogfireClient:
+    def __init__(self, user_token: UserToken) -> None:
+        if user_token.is_expired:
+            raise RuntimeError
+        self.base_url = user_token.base_url
+        self._token = user_token.token
+        self._session = Session()
+        self._session.headers.update({'Authorization': self._token, 'User-Agent': UA_HEADER})
+
+    def _get(self, endpoint: str) -> Response:
+        response = self._session.get(urljoin(self.base_url, endpoint))
+        UnexpectedResponse.raise_for_status(response)
+        return response
+
+    def _post(self, endpoint: str, body: Any | None = None) -> Response:
+        response = self._session.post(urljoin(self.base_url, endpoint), json=body)
+        UnexpectedResponse.raise_for_status(response)
+        return response
+
+    def get_user_organizations(self) -> list[dict[str, Any]]:
+        """Get the organizations of the logged-in user."""
+        try:
+            response = self._get('/v1/organizations')
+        except UnexpectedResponse as e:
+            raise LogfireConfigError('Error retrieving list of organizations') from e
```

Comment:
Looks like it should be something like this:

```suggestion
        response = self._get('/v1/organizations')
        LogfireConfigError.raise_for_status(response, 'Error retrieving list of organizations')
```

Reply:
Actually I think the client should raise a proper `ClientError` exception or similar, which should then be reraised as a `LogfireConfigError`. Because the client is only used during configuration, I left it as is for simplicity (this can be changed if the client gets used outside configuration logic at some point). However, I don't think the `LogfireConfigError` should be aware of HTTP errors as this will make it harder to change in the future if — as I mentioned — we want a proper client exception.